### PR TITLE
Document common issue people might run into when writing Cloud Code

### PR DESCRIPTION
**Pull request by hramos**
We see at least one question related to this common mistake every day in our community forums.

### CODE REVIEW

File: en/common/cloud-code.mdown [hunk 1]
```diff
@@ -10,7 +10,7 @@ Even if you're only familiar with mobile development, we hope you'll find Cloud
 
 ## Getting Started
 
-On the computer you use for development, you will need to install Parse's command line tool. This will let you manage your code in the Parse Cloud. You can learn more about the feautres of the command line tool in the [Command Line Tool guide](#command-line).
+On the computer you use for development, you will need to install Parse's command line tool. This will let you manage your code in the Parse Cloud. You can learn more about the features of the command line tool in the [Command Line Tool guide](#command-line).
 
 ### Instllaling the Command Line Tool
```
Comment (gfosco) [inline en/common/cloud-code.mdown]: typo: Installing

File: en/common/cloud-code.mdown [hunk 2]
```diff
@@ -379,6 +379,46 @@ Cloud functions will be killed after 15 seconds of wall clock time. `beforeSave`
...
+  });
+
+  response.success(results); // Response: "Retrieval all Posts:\n"
+});
+```
+
+This is a common mistake that can be easily avoided. Make sure that you only call `success` once all of your network queries have returned data, such as in the following example:
+
+```js
+Parse.Cloud.define("ThisFunctionWillNotReturnAllDataAsExpected", function(request, response) {
```
Comment (gfosco) [inline en/common/cloud-code.mdown]: This function will return all data as expected, i think..

File: en/common/cloud-code.mdown [hunk 3]
```diff
@@ -379,6 +379,46 @@ Cloud functions will be killed after 15 seconds of wall clock time. `beforeSave`
...
+  var results = "Retrieved all Posts:\n"
+  var query = new Parse.Query("Post");
+
+  // This query.find() is unlikely to finish before response.success() is called.
```
Comment (gfosco) [inline en/common/cloud-code.mdown]: this comment also included from the previous example.

File: en/common/cloud-code.mdown (hunk 1 above)
Comment (hramos) [inline en/common/cloud-code.mdown]: Good catch

File: en/common/cloud-code.mdown (hunk 2 above)
Comment (hramos) [inline en/common/cloud-code.mdown]: Ooops, fixed.

File: en/common/cloud-code.mdown (hunk 3 above)
Comment (hramos) [inline en/common/cloud-code.mdown]: Fixed

**Pull request merged**